pyrobosim/examples/test_world.py

```diff
@@ -0,0 +1,150 @@
+
+        desk = TestWorldModeling.world.add_location("desk", "bedroom", name="study_desk", pose=Pose(x=3.15, y=3.65, z=0.0, yaw=0.0))
+        assert len(TestWorldModeling.world.locations) == 2
+        assert TestWorldModeling.world.get_location_by_name("study_desk") == desk
+
+
+    @staticmethod
+    @pytest.mark.dependency(depends=
+        ["TestWorldModeling::test_create_world_default", "TestWorldModeling::test_create_room", 
+        "TestWorldModeling::test_create_hallway", "TestWorldModeling::test_create_location"])
+    def test_create_object():
+        """ Tests adding objects to a location"""
+
+        TestWorldModeling.world.add_object("apple", "table0")
+        TestWorldModeling.world.add_object("apple", "study_desk")
+        assert len(TestWorldModeling.world.objects) == 2
+
+        # second apple
+        test_obj = TestWorldModeling.world.objects[1]
+        assert isinstance(test_obj, Object)
+        assert TestWorldModeling.world.get_object_by_name("apple1") == test_obj # Automatic naming
+
+
+    @staticmethod
+    @pytest.mark.dependency(depends=["TestWorldModeling::test_create_object"])
+    def test_remove_object():
+        """ Tests deleting objects from the world """
+
+        assert TestWorldModeling.world.remove_object("apple1") is True
+        assert len(TestWorldModeling.world.locations) == 2
+        assert TestWorldModeling.world.get_object_by_name("apple1") is None
+        assert TestWorldModeling.world.objects[0].name == "apple0"
+
+    @staticmethod
+    @pytest.mark.dependency(depends=["TestWorldModeling::test_create_location"])
+    def test_remove_location():
+        """ Tests removing a location from the world """
+
+        assert TestWorldModeling.world.remove_location("study_desk") is True
+        assert len(TestWorldModeling.world.locations) == 1
+        assert TestWorldModeling.world.get_location_by_name("study_desk") is None # Raises a warning
+
+    @staticmethod
+    @pytest.mark.dependency(depends=["TestWorldModeling::test_create_hallway"])
+    def test_remove_hallway():
+        """ Tests removing a hallway """
+
+        hallways = TestWorldModeling.world.get_hallways_from_rooms("kitchen", "bedroom")
+        assert len(hallways) == 1
+        assert isinstance(hallways[0], Hallway)
+
+        assert TestWorldModeling.world.remove_hallway(hallways[0]) is True
+        hallways = TestWorldModeling.world.get_hallways_from_rooms("kitchen", "bedroom")
+        assert len(hallways) == 0
+  
+    @staticmethod
+    @pytest.mark.dependency(depends=["TestWorldModeling::test_create_room"])
+    def test_remove_room():
+        """ Tests deleting rooms """
+
+        assert TestWorldModeling.world.remove_room("bedroom") is True
+        assert len(TestWorldModeling.world.rooms) == 1
+        assert TestWorldModeling.world.rooms[0].name == "kitchen"
+
+    @staticmethod
+    def test_heirarcial_cleanup():
```

Comment:
Should there be dependencies in this test?

Reply:
Ah yes!
Was supposed to add those in !